### ISSUE TITLE
SEO: Check how Google crawler sees HTML version of the book

### ISSUE THREAD

**Issue by ZSvedic** [ZSvedic/EBH-book#19]
*Problem:*
We want Google to index separate chapters of EBH book, although it is one HTML page.

*Solution:*
Check how Google sees current EBH html and is it able to index separate chapters?
Tools:
* [Google URL inspection tool](https://support.google.com/webmasters/answer/9012289?hl=en)
* Article [How to Look at Your Website the Way Google Does](https://neilpatel.com/blog/the-way-google-scans/)
If Google doesn't index separate chapters correctly, propose a solution (sitemap, semantic tags, etc.)

**Comment by reinaldoarrosi** [comment]
I did some investigation regarding this issue. Here's what I found:

According to [this article](https://www.searchenginepeople.com/blog/anchor-links.html), Google will index all chapters and anchors to these chapters when crawling the book page. This is because we have all content in a single HTML file with anchors that link to specific chapters/sub-chapters. Google _may_ also show an anchor link (such as https://testdome.com/evidence-based-hiring/book/ebh.html#chapter-4-reward-vs-effort) in its search result page.

But this doesn't mean that Google sees multiple pages. To Google, the book is a single page with anchors that take the user to different places within the page. 

Here's what I found about making a single page appear as multiple pages to Google:

1. We're using anchors to navigate between chapters. By default, Google will not consider anchors when as separate pages when indexing a page. Until 2018 Google recommended using "hashbang routing" (aka #!) to solve this, but they removed this feature in summer 2018.

2. Google now recommends using the history API to do client-side routing. This enables links such as <a href="/chapter1"> to be handled via JS, without triggering an HTTP request to "/chapter1". This way googlebot sees <a> tags with correct href attributes and is able to index each chapter as a separate page. Problem is that Pandoc generates the table-of-contents using # (ex.: <a href="#chapter1">) and we don't have control over it (not even using LUA filters).
